teamB/jeonghun/app.js

```diff
@@ -0,0 +1,123 @@
+require('dotenv').config();
+const http = require ('http');
+const express = require('express');
+const cors = require('cors');
+const morgan = require('morgan');
+
+const {DataSource} = require('typeorm');
+
+const appDataSource = new DataSource({
+    type: process.env.TYPEORM_CONNECTION,
+    host: process.env.TYPEORM_HOST,
+    port: process.env.TYPEORM_PORT,
+    username: process.env.TYPEORM_USERNAME,
+    password: process.env.TYPEORM_PASSWORD,
+    database: process.env.TYPEORM_DATABASE
+});
+
+appDataSource.initialize()
+ .then(() => {
+  console.log('Data Source has been initialized!')  
+ })
+ .catch((error) => {
+  console.error('Error during Data source initialization', error);
+ });
+
+const app = express()
+
+const server = http.createServer(app)
+const PORT = process.env.PORT;
+
+app.use(express.json());
+app.use(cors());
+app.use(morgan('dev'));
+
+app.get('/ping', function (req, res, next) {
+    res.status(201).json({message: 'pong'});
+});
+
+app.post('/user/sign', async (req, res, next) => {
+  const { name, email, profileImage, password } = req.body
+
+  // console.log(req)
+
+  await appDataSource.query(
+    `INSERT INTO users(
+      name,
+      email,
+      profile_image,
+      password
+      ) VALUES (?, ?, ?, ?);
+      `,
+      [ name, email, profileImage, password ]
+    );
+     res.status(201).json({ message : 'userCreated'});
+    })
+
+    app.post('/post', async (req, res, next) => {
+      const { title, content } = req.body
+    
+      // console.log(req)
+    
+      await appDataSource.query(
+        `INSERT INTO posts(
+          title,
+          content
+          ) VALUES (?, ?);
+          `,
+          [ title, content ]
+        );
+         res.status(201).json({ message : 'successfully created'});
+        })    
+
+        app.post('/like', async (req, res, next) => {
+          const { userid, postid } = req.body
+        
+          await appDataSource.query(
+            `INSERT INTO likes(
+              user_id,
+              post_id
+              ) VALUES (?, ?);
+              `,
+              [ userid, postid ]
+            );
+             res.status(201).json({ message : 'successfully created'});
+            })   
+
+app.get('/user', async (req, res) => {
+  await appDataSource.manager.query(
+    `SELECT
+         b.id,
+         b.title,
+         b.description,
+         b.cover_image
+        FROM users b`
+    ,(err, rows) => {
+         res.status(200).json(rows);
+    })
+});
```

Comment:
- RESTfull에서 더 자세히 배우겠지만, 목록을 보는 api 같은 경우 복수를 많이 사용합니다.
- SQL을 사용할 때 적용되는 테이블이 하나일 경우 별칭을 굳이 안 사용해도 되고 만약 사용한다면 어떤 테이블의 별칭인지 알 수 있도록 해당 테이블의 앞 스펠링을 많이 사용합니다. 예를 들어 users의 대한 별칭이면 b가 아닌 u를 사용하는 것이 좋습니다.
```suggestion
app.get('/users', async (req, res) => {
  await appDataSource.manager.query(
    `SELECT
         b.id,
         title,
         description,
         cover_image
        FROM users`
    ,(err, rows) => {
         res.status(200).json(rows);
    })
});
```